### DISCUSSION TITLE
How do I include next variables in .modules.scss files?

### DISCUSSION THREAD

**Discussion by sansyrox** [vercel/next.js#20276] [Help] [answered]
I have a `variables.scss` file where I declare all the sass variables.

But whenever I want to use those variables in the .module.scss files, I have to reference them using `@import ../../variables.scss`

Is there a way to make them available throughout the scss files?

I tried importing them in the globals .scss file but it didn't work and I also went through a lot of discussions on GitHub but I can't seem to understand it.

Thanks in advance! 😄

**Comment by jamesmosier** [comment, marked as answer]
Hi @sansyrox. You can include this in your `next.config.js` which will import your variables into any `scss` file in your project:

```js
module.exports = {
  sassOptions: {
    prependData: `
        /// Scss code that you want to be
        /// prepended to every single scss file.
        @import 'path/to/your/variables';
      `,
  }
}
```

**Reply by sansyrox** [reply]
Hi @jamesmosier , thank you for reverting!

 I saw this approach but the variables file is a fairly large file. I don't want to write it in JavaScript.

Is there a way to extend the webpack with this extension: https://www.npmjs.com/package/sass-resources-loader

I can't seem to figure out a way to extend webpack in next.config.js.

Sorry, if this has been answered already 😓

**Reply by jamesmosier** [reply]
> I don't want to write it in JavaScript.

You don't need to write it in JavaScript, the `@import 'path/to/your/variables';` is for a file called `variables.scss` that contains SCSS variables - no JavaScript needed.

Here's the PR for reference https://github.com/vercel/next.js/pull/12277

**Reply by sansyrox** [reply]
Oh right. I misread. Thank you so much!!! ❇️

**Reply by sansyrox** [reply]
@jamesmosier , one last question. Since my imports look like `@import '../variables'` and `@import '../styles/variables'` , i.e. they are different based on the location of the sass files. 

Can I also prepend different data depending on the folder in which the file belongs?

**Reply by jamesmosier** [reply]
The `prependData` option in your `next.config.js` will be a global option. So whatever you import there will be available in all files. So there is no way to select a different variables depending on what file.

**Reply by sansyrox** [reply]
Oh okay makes sense. But essentially we'll be having an `import 'variables.scss'` in the variables file too, right?
